### ISSUE TITLE
Allow PersistentTopics#updatePartitionedTopic to also delete partitions

### ISSUE THREAD

**Issue by ConcurrencyPractitioner** [apache/pulsar#4152]
### What is the problem?
So far, in the current Pulsar REST API available to users. ```PersistentTopics#updatePartitionedTopic()``` is able to increase the number of partitions, however it is unable to delete any partitions. This behavior should be modified so that the number of partitions could be deleted as well.

### Possible Solutions
Partitions could be forcefully deleted in Pulsar API. We could delete partitions in this manner, even if a subscription was attached. 

This would allow users to scale the number of partitions in a more flexible manner, since only allowing them to increase the number of partitions is very annoying.

**Comment by ConcurrencyPractitioner** [comment]
Oh @merlimat your thoughts on this?

**Comment by sijie** [comment]
@ConcurrencyPractitioner 

I don't think force deleting partition is the right direction. because it will cause data loss.

I think the right approach should be waiting for all the data in the old partitions to be consumed before deleting the old partitions. Hence:

- the broker has to maintain the metadata before partitions scaling-down and the metadata afterwards.
- the client needs to be able to fetch such metadata for choosing the right partition to read.

**Comment by ConcurrencyPractitioner** [comment]
@sijie Oh, in that case, wouldn't we have to stop the flow of new records/messages into the ledgers? From my interpretation, I think from the approach you described, what we should do is basically stop all flow of new data from Producer to the partitions we wish to delete, finish processing the data that is still there, then delete them. 

I think when we delete partitions, the most straightforward approach is to delete the last N-K partitions if there was originally N partitions, and the user specifies us for K partitions, where K < N. The partitions which are to be deleted I think would be made read-only once the user requests them to be deleted.

**Comment by tisonkun** [comment]
Closed as stale. Please create a new issue if it's still relevant to the maintained versions.

It seems this proposal got rejected.
